v5/ooapi.v5.UnitTests/Controllers/AcademicSessionsControllerTests.cs

```diff
@@ -0,0 +1,124 @@
+using Microsoft.AspNetCore.Mvc;
+using ooapi.v5.Controllers;
+using ooapi.v5.core.Services.Interfaces;
+using ooapi.v5.core.Utility;
+using ooapi.v5.Models;
+using ooapi.v5.Models.Params;
+
+namespace ooapi.v5.UnitTests.Controllers;
+
+public class AcademicSessionsControllerTests
+{
+    private readonly IFixture _fixture = new Fixture();
+
+    [Test]
+    public void AcademicSessionsGet_ByPrimaryCode_ReturnsAcademicSessions()
+    {
+        //arrange
+        var sut = CreateSut(out var academicSessionsService);
+        var primaryCodeParam = _fixture.Create<PrimaryCodeParam>();
+        FilterParams? filterParams = null;
+        PagingParams? pagingParams = null;
+        var academicSessionType = _fixture.Create<string?>();
+        var parent = _fixture.Create<Guid?>();
+        var year = _fixture.Create<Guid?>();
+        var sort = _fixture.Create<string?>();
+
+        var response = new Pagination<AcademicSession>();
+
+        DataRequestParameters? dataRequestParameters = null;
+
+        academicSessionsService.GetAll(Arg.Do<DataRequestParameters>(x => dataRequestParameters = x), out var errorResponse, academicSessionType).Returns(response);
+
+        //act
+        var result = sut.AcademicSessionsGet(primaryCodeParam, filterParams, pagingParams, academicSessionType, parent, year, sort) as OkObjectResult;
```

Comment:
Moet de definitie van filterParams niet ook nullable zijn in de controller AcademicSessiosGet method?. Als dat daar niet kan, dan zou ik ze hier ook niet als nullable definieren.

Dit geldt op meer plekken overigens.